commerce/domain/shipping/splitshipment/SplitShipmentSample.cls

```diff
@@ -0,0 +1,129 @@
+/**
+ * @description Custom Split Shipment Service sample.
+ */
+global class SplitShipmentSample extends CartExtension.SplitShipmentService {
+
+    public virtual override void arrangeItems(CartExtension.ItemArrangementRequest request) {
+        CartExtension.Cart cart = request.getCart();
+
+        Iterator<CartExtension.ItemArrange> iterator = request.getItemArrangeList().iterator();
+        while (iterator.hasNext()) {
+            CartExtension.ItemArrange itemArrange = iterator.next();
+            arrangeItemsByProductConveyability(cart, itemArrange);
+        }
+    }
+
+    /**
+     * @description Split up the shipment by conveyable and non-conveyable products. Products that are non-conveyable
+     * cannot go on the conveyor belt at the warehouse because they are too big, bulky, and/or heavy. This means that
+     * these non-conveyable items may require additional packaging and will take longer to get onto a delivery truck.
+     * @param cart Holds details about a Cart
+     * @param itemArrange Holds information on how the item should be arranged
+     */
+    private void arrangeItemsByProductConveyability(CartExtension.Cart cart, CartExtension.ItemArrange itemArrange) {
+        // Get conveyable info for the product
+        List<Product2> products = [SELECT IsConveyable__c FROM Product2 WHERE Id =: itemArrange.getProductId() LIMIT 1];
```

Comment:
is this picking any conveyable product?

Reply:
In the unit test setup, I created 3 products in the `Product2` entity, 1 of which is non-conveyable. See [this method](https://github.com/forcedotcom/commerce-extensibility/pull/44/files#diff-0103db420f68fbe11d86b7ee488593e8dd5d4a0d05faa615e09ba5a8b334855bR137).

Reply:
Why doing this query for each arrangeItem instead of doing one query for all products in arrangeItemList?
Also, this is a use case to only sort items with products limited by arrangeItems. Most customers who want to do auto-split would just iterate through the cart items and split based on the custom attributes. I guess you are still trying to show case an instructed split here?

Reply:
Ahh yes good call lol

Reply:
@wei-liu-salesforce this has been addressed in the latest commit.